File: 백준/Silver/9375. 패션왕 신해빈/패션왕 신해빈.py
import sys
input = sys.stdin.readline

T = int(input())
for t in range(T):
    n = int(input())
    wear = {}
    for i in range(n):
        name, type_ = input().split()
        if type_ in wear:
            wear[type_].append(name)
        else:
            wear[type_] = [name]
    cnt = 1
    for j in wear.values():
        cnt *= len(j) + 1
    print(cnt - 1)

File: 백준/Silver/9375. 패션왕 신해빈/패션왕 신해빈.swift
import Foundation
let testCase = Int(readLine()!)!
for _ in 0..<testCase {
    var dict = [String: [String]]()
    var count = 1
    let n = Int(readLine()!)!
    for _ in 0..<n {
        let input = readLine()!.split(separator: " ").map(\.description)
        if dict[input[1]] != nil {
            dict[input[1]]?.append(input[0])
        } else {
            dict[input[1]] = [input[0]]
        }
    }
    for value in dict.values {
        count *= value.count + 1
    }
    print(count - 1)
}